libnmstate/schemas/interfaces.json

```diff
@@ -10,21 +10,89 @@
             "items": {
                 "title": "Interface",
                 "type": "object",
-                "properties": {
-                    "name": { "type": "string" },
-                    "description": { "type": "string" },
-                    "type": {
-                        "type": "string",
-                        "enum": ["ethernet", "bond", "vlan"]
-                    },
-                    "state": {
-                        "type": "string",
-                        "enum": ["absent", "up", "down", "unknown"]
+                "oneOf": [
+                    {
+                        "allOf": [
+                            {
+                                "properties": {
+                                    "type": {
+                                        "type": "string",
+                                        "enum": ["ethernet", "bond", "vlan"]
+                                    }
+                                }
+                            },
+                            { "$ref": "#/definitions/interface-rw" },
+                            { "$ref": "#/definitions/interface-ro" }
+                        ]
                     }
-                },
+                ],
                 "required": ["name"]
```

Comment:
you require name byt it is not a property of this object

Reply:
It needs to be located in the same level as the "properties" and works on it.